### PR TITLE
Test refactoring sync_mode

### PR DESCRIPTION
**Pull request by benrudolph**
Taking a stab at getting casexml tests down. This one refactors sync_mode just a bit by moving things that can be done in SetUpClass to there instead of setUp.

Before:
```

----------------------------------------------------------------------
Ran 68 tests in 108.208s
```
After:
```

----------------------------------------------------------------------
Ran 68 tests in 54.738s
```
@dimagi/test-team

### CODE REVIEW

**Review by snopoke — APPROVED** [review]
:1st_place_medal:

**Comment by benrudolph** [comment]
Realized it's especially brutal to do extra work in the `setUp` and `tearDown` call because not only does this get called once per test, but for those decorated with `run_with_all_backends` it gets called twice (i believe)

**Comment by benrudolph** [comment]
For all of casexml:

Before:
```
----------------------------------------------------------------------
Ran 390 tests in 353.083s
```
After:
```
----------------------------------------------------------------------
Ran 390 tests in 147.810s
```

**Comment by snopoke** [comment]
Yea, that's the same as I found with this one https://github.com/dimagi/commcare-hq/pull/14838

**Review by kaapstorm — APPROVED** [review]

**Comment by emord** [comment]
Would it be possible to have `run_with_all_backends` decorate the class instead of individual tests? Or does that not work becasue run_with_all_backends logs two setup calls per test, instead of splitting them into two tests?

File: corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py [hunk 1]
```diff
@@ -42,21 +42,26 @@ class SyncBaseTest(TestCase):
     """
     Shared functionality among tests
     """
+    @classmethod
+    def setUpClass(cls):
+        super(SyncBaseTest, cls).setUpClass()
+        delete_all_users()
```
Comment (emord) [inline corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py]: Does it still work if you remove this line?

**Review by emord — APPROVED** [review]

**Pull request merged**